add `kubernixos build` and move kubeval to build phase

johanot opened the pull request:
- refactor: module changes + drop vendored kubeval-schema
- added: kubernixos build subcommand

Commit:
- refactor: module changes + drop vendored kubeval-schema

- added: kubernixos build subcommand

Commit:
lib: we don't need to be able to pass in nixpkgs/lib

Commit:
module: no default schema package please, we want things explicit

srhb reviewed (commented):
~What happens here if a json string includes a single quote?~ didn't get this attached to code somehow. Fun

srhb commented on lib/kubernixos.nix:
```diff
@@ -19,26 +20,30 @@ let
+  build =
+      pkgs.runCommand "kubernixos-${kubernixos}" { nativeBuiltInputs = [pkgs.kubeval cfg.schemas]; } ''
+          mkdir -p $out
+          ${lib.concatStringsSep "\n" (lib.mapAttrsToList (n: i: "echo '${builtins.toJSON i}' >$out/${n}.json") items)}
```
What happens here if a json string contains a single quote? Accept build failure and move on?

srhb replied:
Alternatively one could build `writeText` each json snippet and cat them together, or create both the individuals and the sum in Nix before writing it out. No reason to go through bash strings.

johanot replied:
Have a look at: a245a1d4db17ca810cbc4c595eb5f8c2d26b04df

srhb replied:
:ok_hand:

Commit:
- build all manifests with pkgs.writeText and symlink from kubernixos drv

- kubeval 0.15 accepts k8s Lists now, so just pass it the final generated list, which name is changed from "all-validated.json" to "kubernixos.json"
